### ISSUE TITLE
MediaSpace URL not included in the API response

### ISSUE THREAD

**Issue by janklimo** [kaltura/server#6536]
Looking at the `media.list` endpoint response, none of the returned URLs can be used to take the user to the content in MediaSpace. There is `dataUrl`:

```
https://cdnapisec.kaltura.com/p/2351761/sp/235176100/playManifest/entryId/0_77e1jtte/format/url/protocol/https
```

which, by the way, is not found, and it doesn't resemble the matching MediaSpace link:

```
http://2351761.mediaspace.kaltura.com/media/Toward+a+Better+Front+End+ArchitectureA+Elm+-+Codemash+2017/0_77e1jtte/82781251
```

 I believe you can take the returned media object id and compose the URL yourself, like:

```
http://2351761.mediaspace.kaltura.com/media/0_77e1jtte
```

which takes you to the same media but that doesn't seem to be the optimal solution. What is the recommended practice? Thank you!

**Comment by kaltura-hooks** [comment]
Hi @janklimo,

Thank for you reporting an issue and helping improve Kaltura!

To get the fastest response time, and help the maintainers review and test your reported issues or suggestions, please ensure that your issue includes the following (please comment with more info if you have not included all this info in your original issue):
 - Is the issue you're experiencing consistent and across platforms? or does it only happens on certain conditions?
    please provide as much details as possible.
 - Which Kaltura deployment you're using: Kaltura SaaS, or self-hosted?
	If self hosted, are you using the RPM or deb install?
 - Packages installed.
	When using RPM, paste the output for:
```
	# rpm -qa \"kaltura*\"
```	
	For deb based systems:
```	
	# dpkg -l \"kaltura-*\"
```
 - If running a self hosted ENV - provide the MySQL server version used
 - If running a self hosted ENV - is this a single all in 1 server or a cluster?
 - If running a self hosted ENV, while making the problematic request, run:
```
	# tail -f /opt/kaltura/log/*.log /opt/kaltura/log/batch/*.log | grep -A 1 -B 1 --color \"ERR:\|PHP\|trace\|CRIT\|\[error\]\"
```
   and paste the output.
 - When relevant, provide any screenshots or screen recordings showing the issue you're experiencing.

For general troubleshooting see:
https://github.com/kaltura/platform-install-packages/blob/Jupiter-10.13.0/doc/kaltura-packages-faq.md#troubleshooting-help

If you only have a general question rather than a bug report, please close this issue and post at:
http://forum.kaltura.org

Thank you in advance,

**Comment by jessp01** [comment]
Hello @janklimo,

The reason media.list [or any other endpoint for that matter] does not reference MediaSpace is that MediaSpace and the Kaltura Server are not tightly coupled. 
That is, in order to use MediaSpace, you must have a partner on a Kaltura Server [be it on our SaaS or on a self hosted instance in your own data centre] but you can have a fully functional Kaltura partner without a MediaSpace instance. 

In other words: MediaSpace is an application that relies on the Kaltura Server and uses its API to get things done but the Kaltura Server does not require MediaSpace.

In light of the above, a given partner can have a MediaSpace instance on our SaaS, self hosted on premise or, no MediaSpace instance at all.

I hope that explains it. Do let me know should you have additional questions.

**Comment by janklimo** [comment]
@jessp01 Thank you very much for your quick response!

Would you recommend I build the URL myself as suggested above?

One thing I noticed is that both 

```
http://2351761.mediaspace.kaltura.com/media/id/0_77e1jtte
```

and

```
http://2351761.mediaspace.kaltura.com/media/0_77e1jtte
```

work. Which one is more idiomatic/preferred?

Thank you again for your support!

**Comment by jessp01** [comment]
Hi @janklimo,

First - most welcome:)
As to what is the preferred method, actually, there are 3 formats that currently work:
```
$BASE_URL/media/$ENTRY_ID
$BASE_URL/media/id/$ENTRY_ID
$BASE_URL/media/$ENTRY_NAME/$ENTRY_ID
```
For the third, URL encoding should be used, of course as the name may contain spaces or non-ASCII chars.
I personally like the third best simply because it gives some indication as to the entry I'm about to view but I think that's just me:)

@gonenradai - do you feel any one of these formats is preferable? or, to put it a different way: that one of these may no longer be supported in future versions?

**Comment by gonenradai** [comment]
The preferred method is the third one above.

The $ENTRY_NAME is not a must, and can be replaced with any single character.

In fact, KMS has a configuration that lets the admin choose between having the actual entry name (encoded, as you said) in the URL or not.
Having it in the URL benefits in better SEO.
Some customers preferred not to have the name in the URL to avoid potential "metadata leakage", so this is the reason for the configuration.

The second format above (with "/id/") works just because KMS doesn't really care what comes between the '/media/' which is the first parameter to the application that controls the routing decision, and the $ENTRY_ID which is the third parameter that is really taken as the entry ID.

**Comment by janklimo** [comment]
Thank you so much for your inputs! Closing 👍

**Comment by jessp01** [comment]
Hi @janklimo,

As a side note. for additional questions, please post at forum.kaltura.org since we prefer to use the GitHub issues section for bug reports only.

Thank you,
